### PR TITLE
i18n [nfc]: Adjust "page title" string descriptions for clarity

### PR DESCRIPTION
**Pull request by gnprice**
Some of these were a bit hard to parse.

Also make them somewhat more parallel to each other.

Prompted by: https://github.com/zulip/zulip-flutter/pull/1076#discussion_r1874151721

### CODE REVIEW

File: assets/l10n/app_en.arb [hunk 1]
```diff
@@ -367,7 +367,7 @@
   },
   "loginAddAnAccountPageTitle": "Add an account",
   "@loginAddAnAccountPageTitle": {
-    "description": "Page title for screen to add a Zulip account."
+    "description": "Title for page to add a Zulip account."
```
Comment (PIG208) [inline assets/l10n/app_en.arb:370]: Is it intentional that we omit "the" before "page" here?

Comment (gnprice) [reply assets/l10n/app_en.arb:370]: It's intentional in the sense that lots of other descriptions are in this headline-ese anarthrous (i.e. without articles) style, and I didn't feel this was the time to try to unify everything in that respect :slightly_smiling_face:

**Review by PIG208 — COMMENTED** [review]
Thanks! Feel free to merge it. I will rebase #1076 on top of this for now.

**Comment by chrisbobbe** [comment]
LGTM, merging.
